### ISSUE TITLE
How to Quantize Qwen2.5-VL on image calibration data

### ISSUE THREAD

**Issue by Zhiy-Zhang** [ModelCloud/GPTQModel#1781]
The following code which I tested:

```
messages = [
    {
        "role": "user",
        "content": [
            {
                "type": "image",
                "image": "https://qianwen-res.oss-cn-beijing.aliyuncs.com/Qwen-VL/assets/demo.jpeg",
            },
            {"type": "text", "text": "Describe this image."},
        ],
    }
]



quantize_config = QuantizeConfig(
    bits=4,  # quantize model to 4-bit
    group_size=128,  # it is recommended to set the value to 128
)

# load un-quantized model, by default, the model will always be loaded into CPU memory
model = GPTQModel.load(pretrained_model_id, quantize_config)

# quantize model, the calibration_dataset should be list of dict whose keys can only be "input_ids" and "attention_mask"
model.quantize(messages)

# save quantized model
model.save(quantized_model_id)
```

Is the above implementation correct？

**Comment by Qubitium** [comment]
@Zhiy-Zhang Please check our tests/models ci tests for per model sample usage.

**Comment by Zhiy-Zhang** [comment]
@Qubitium 
Thank you for your reply. I plan to use my own dataset for quantization, but I'm not entirely clear about the specific workflow. I'm referring to this test file “ test_qwen2_5_vl.py”, 
After reading the source code：https://github.com/ModelCloud/GPTQModel/blob/main/tests/models/model_test.py#L205, I understand that following this workflow, can I achieve quantization by simply modifying this part to replace it with my own dataset?

**Comment by Qubitium** [comment]
@Zhiy-Zhang You are correct. Follow the same workflow but tweak/inject your own dataset plus quant params and you are all good

**Comment by Zhiy-Zhang** [comment]
@Qubitium Thank you for your reply.

**Comment by Zhiy-Zhang** [comment]
@Qubitium I tried modifying the dataset, but I encountered the following error when running it. It seems that the device of grid_thw is CPU. Is this expected? My GPTQModel was compiled from the source code on the main branch, and the Transformers version is 4.56.1.

<img width="1886" height="922" alt="Image" src="https://github.com/user-attachments/assets/8d0c3176-c29b-4b1c-b88b-5c4f4e1eefb5" />

**Comment by Zhiy-Zhang** [comment]
@Qubitium  If the device of cu_seqlens is set to CUDA, quantization can proceed normally, but the following error occurs during execution.

<img width="1748" height="760" alt="Image" src="https://github.com/user-attachments/assets/6ec53163-68a0-44d4-b93f-008ab0978328" />

The device appears to be  “device(type='cuda')”
I'm not sure if there's an issue with my environment; could you please provide some troubleshooting suggestions?

**Comment by Qubitium** [comment]
> @Qubitium  If the device of cu_seqlens is set to CUDA, quantization can proceed normally, but the following error occurs during execution.
> 
> <img width="1748" height="760" alt="Image" src="https://github.com/user-attachments/assets/6ec53163-68a0-44d4-b93f-008ab0978328" />
> 
> The device appears to be  “device(type='cuda')”
> I'm not sure if there's an issue with my environment; could you please provide some troubleshooting suggestions?


Please code to src and var usage of cu_seqlens.

**Comment by Zhiy-Zhang** [comment]
@Qubitium 
I modified two parts to make the entire training process run smoothly.
1、https://github.com/ModelCloud/GPTQModel/blob/main/gptqmodel/looper/module_looper.py#L125：

<img width="1160" height="220" alt="Image" src="https://github.com/user-attachments/assets/cf5c518b-1614-43ce-9cfb-5d5c8e0ff4c4" />
2、https://github.com/ModelCloud/GPTQModel/blob/main/gptqmodel/models/loader.py#L534

<img width="1054" height="250" alt="Image" src="https://github.com/user-attachments/assets/0341cd35-a578-45a6-8f50-d723751efae4" />

**Comment by Qubitium** [comment]
@LRL-ModelCloud check this. By default we should pick the best GPU, Cuda for this user in this call. Something happened to cause this regress?

**Comment by Zhiy-Zhang** [comment]
@Qubitium I only used one data sample for quantization, and then used the same sample for testing, but found that the results overflowed. Is this expected?

**Comment by Qubitium** [comment]
@Zhiy-Zhang Previous device errors should be fixed in pr #1783

**Comment by Qubitium** [comment]
> @Qubitium I only used one data sample for quantization, and then used the same sample for testing, but found that the results overflowed. Is this expected?

What do you mean overflowed? Please provide stack traces for every exception especially NaN type errors

**Comment by Zhiy-Zhang** [comment]
@Qubitium 

<img width="2026" height="588" alt="Image" src="https://github.com/user-attachments/assets/92939a60-0a80-410d-adb4-f9cbe87219c4" />

my model:
https://github.com/bytedance/UI-TARS

**Comment by Qubitium** [comment]
> @Qubitium 
> 
> <img width="2026" height="588" alt="Image" src="https://github.com/user-attachments/assets/92939a60-0a80-410d-adb4-f9cbe87219c4" />
> 
> my model:
> https://github.com/bytedance/UI-TARS
> 

That is not overflow..it is just bad output due to bad quantization likely due to you using one single input/calibration sample. Check the loss values during quant stage. The lower the better, usually.

**Comment by Qubitium** [comment]
@Zhiy-Zhang  I am going to close this issue and the new threads are unrelated to the original issue. Please open a new issue if you have any other questions.
